### PR TITLE
[backport] [special] add `cupyx.scipy.special.log_softmax`

### PR DESCRIPTION
**Pull request by takagi**
Backport of #6823 by @khushi-411

### CODE REVIEW

**Comment by takagi** [comment]
blocked by https://github.com/cupy/cupy/pull/6894

**Comment by takagi** [comment]
blocked by #6963.

File: tests/cupyx_tests/scipy_tests/special_tests/test_log_softmax.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import pytest
+
+import cupy
+from cupy import testing
+import cupyx.scipy.special  # NOQA
+
+import scipy.special  # NOQA
```
Comment (kmaehashi) [inline tests/cupyx_tests/scipy_tests/special_tests/test_log_softmax.py:7]:
```suggestion
```

**Pull request merged**